image_transport/CMakeLists.txt

```diff
@@ -1,69 +1,123 @@
-    roslib
-    sensor_msgs
-)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(Boost REQUIRED)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  DEPENDS message_filters pluginlib rosconsole roscpp roslib sensor_msgs
-)
+find_package(ament_cmake REQUIRED)
+
+find_package(message_filters REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(tinyxml_vendor REQUIRED)
+find_package(TinyXML REQUIRED)
 
-include_directories(include ${catkin_INCLUDE_DIRS})
+include_directories(include)
 
-# Build libimage_transport
+# Build image_transport library
 add_library(${PROJECT_NAME}
+  SHARED
   src/camera_common.cpp
+  src/publisher.cpp
+  src/subscriber.cpp
+  src/single_subscriber_publisher.cpp
   src/camera_publisher.cpp
   src/camera_subscriber.cpp
   src/image_transport.cpp
-  src/publisher.cpp
-  src/single_subscriber_publisher.cpp
-  src/subscriber.cpp
 )
-add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
-target_link_libraries(${PROJECT_NAME}
-  ${Boost_LIBRARIES}
-  ${catkin_LIBRARIES}
+
+ament_target_dependencies(${PROJECT_NAME}
+  "message_filters"
+  "pluginlib"
+  "rclcpp"
+  "sensor_msgs"
+  "TinyXML"
 )
 
-# Build libimage_transport_plugins                                      
-add_library(${PROJECT_NAME}_plugins src/manifest.cpp src/raw_publisher.cpp)
-target_link_libraries(${PROJECT_NAME}_plugins ${PROJECT_NAME})
+target_compile_definitions(${PROJECT_NAME} PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
 
-install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_plugins
-        DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        COMPONENT main
-)
-install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+# Build image_transport_plugins library (raw)
+add_library(${PROJECT_NAME}_plugins
+  SHARED
+  src/manifest.cpp
 )
+target_link_libraries(${PROJECT_NAME}_plugins ${PROJECT_NAME})
 
-# add two execs
+# Build list_trasnports
+add_executable(list_transports src/list_transports.cpp)
+target_link_libraries(list_transports ${PROJECT_NAME})
+
+# Build republish
 add_executable(republish src/republish.cpp)
 target_link_libraries(republish ${PROJECT_NAME})
 
-add_executable(list_transports src/list_transports.cpp)
-target_link_libraries(list_transports
-  ${PROJECT_NAME}
-  ${catkin_LIBRARIES}
+# Install plugin descriptions
+pluginlib_export_plugin_description_file(image_transport default_plugins.xml)
+
+# Install libraries
+install(
+  TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_plugins
+  LIBRARY DESTINATION lib
 )
 
-install(TARGETS list_transports republish
-        DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+# Install executables
+install(
+  TARGETS list_transports republish
+  RUNTIME DESTINATION share/${PROJECT_NAME}
```

Comment:
Executables should be installed to `DESTINATION lib/${PROJECT_NAME}` to be found by `ros2 run`